database_adapter/src/access.py

```diff
@@ -84,3 +85,44 @@ def parse_to_int(s: str) -> int:
             )
 
             return [i[0] for i in cur.fetchall()]
+
+@access_router.post("/ble_device_detected")
+def post_ble_addr(mac_addr: str = Query(max_length=17)):
+    with main_db_pool.connection() as conn:
+        with conn.cursor() as cur:
+            cur.execute(
+                "INSERT INTO public.ble_stats (mac_addr) "
+                "VALUES (%s) "
+                "RETURNING mac_addr",
+                (mac_addr,)
+            )
+
+            return bool(cur.fetchall()[0])
+
+@access_router.get("/ble_last_devices")
+def ble_last_devices(time: int):
+    with main_db_pool.connection() as conn:
+        with conn.cursor() as cur:
+            cur.execute(
+                "SELECT DISTINCT mac_addr FROM public.ble_stats as ble "
+                "WHERE ble.timestamp >= date_trunc('hour', current_timestamp) "
+                f"+ date_part('minute', current_timestamp)::int / {time} * interval '{time} minute'",
```

Comment:
avoid string interpolation to prevent sql injection - shouldn't be a problem here as time is an int but generally dangerous